File: wwwroot/js/TypeModules/Areas/DepManage/ViewModels/fvmEdit.js
/* ===== 此檔案是自動產生 ===== */
/* ===== 請勿手動變更修改 ===== */

/**
 * 新增/編輯的ViewModel
 * @class
 */
class fvmEdit {
    /**
     * 流水號
     * @type {number}
     */
    ID;
    /**
     * Vue模式
     * @type {number}
     */
    VueMode;
    /**
     * 名稱
     * @type {string}
     */
    Name;
    /**
     * 縮寫
     * @type {string}
     */
    Alias;
    /**
     * 上層部門的ID
     * @type {number}
     */
    ParentID;
    /**
     * 成立日期
     * @type {datetime}
     */
    EstablishDate;

    /** 建構式 */
    constructor () {
        this.ID = 0;
        this.ModelError__ID = '';
        this.VueMode = 0;
        this.ModelError__VueMode = '';
        this.Name = "";
        this.ModelError__Name = '';
        this.Alias = "";
        this.ModelError__Alias = '';
        this.ParentID = 0;
        this.ModelError__ParentID = '';
        this.EstablishDate = "";
        this.ModelError__EstablishDate = '';
    }

    /** 
     * 取得屬性的型別
     * @param {string} prop 屬性名稱
     * @returns {string|Map<string, string>} 屬性的型別
     */
    propertyDictionary (prop = undefined) {
        const propMap = new Map([
            ['ID',            'number'],
            ['VueMode',       'number'],
            ['Name',          'string'],
            ['Alias',         'string'],
            ['ParentID',      'number'],
            ['EstablishDate', 'datetime'],
        ]);
        if (prop == undefined) {
            return propMap;
        }
        return propMap.get(prop);
    }
}
export {
    fvmEdit
}

File: Areas/DepManage/ViewModels/fvmEdit.cs
using System.ComponentModel;

namespace ElectronApp.Areas.DepManage.ViewModels
{
    /// <summary>
    /// 新增/編輯的ViewModel
    /// </summary>
    public class fvmEdit
    {
        /// <summary> 流水號 </summary>
        public long ID { get; set; }

        /// <summary> Vue模式 </summary>
        public int VueMode { get; set; }

        /// <summary> 名稱 </summary>
        [DisplayName("名稱")]
        public string Name { get; set; }

        /// <summary> 縮寫 </summary>
        [DisplayName("縮寫")]
        public string Alias { get; set; }

        /// <summary> 上層部門的ID </summary>
        [DisplayName("上層部門的ID")]
        public int ParentID { get; set; }

        /// <summary> 成立日期 </summary>
        [DisplayName("成立日期")]
        public DateTime EstablishDate { get; set; }

        /// <summary> 所屬人員ID清單 </summary>
        public List<fvmEditUsers> Users { get; set; }
    }

    /// <summary>
    /// 使用者ID
    /// </summary>
    public class fvmEditUsers
    { 
        /// <summary>流水號</summary>
        public long ID { get; set; }
        /// <summary>使用者ID</summary>
        public long UserID { get; set; }
        /// <summary>部門ID</summary>
        public long DepartmentID { get; set; }
        /// <summary>使用者名稱</summary>
        public string Name { get; set; } 
        /// <summary>是否刪除</summary>
        public int IsDelete { get; set; }
    }
}
